File: demos/fortran-python/random.f
      PROGRAM PRINTRANDOM
      
      INTEGER N
      INTEGER SEED
      REAL RANDX
      
      SEED = 128
      RANDX = 0
      
      N = 0
 10   CALL RANDOM(SEED, RANDX)
      PRINT *, RANDX
      N = N + 1
      IF (N.LT.5) GOTO 10
      
      END

* --------------------------------------------
* Psuedo random number generator
* --------------------------------------------

      SUBROUTINE RANDOM(SEED, RANDX)

      INTEGER SEED
      REAL RANDX
      
      SEED = 2045*SEED + 1
      SEED = SEED - (SEED/1048576)*1048576
      RANDX = REAL(SEED + 1)/1048577.0
      RETURN

      END

* --------------------------------------------

File: demos/fortran-python/random.py
def random(seed):
    seed = 2045*seed + 1
    seed = seed - (seed/1048576)*1048576
    randx = (seed + 1)/1048577.0
    return (seed, randx)

seed = 128

for n in range(5):
    (seed, randx) = random(seed)
    print randx
